Fix version orders

JKarlavige opened the pull request:
## What are you changing in this pull request and why?

### Issue
With the current versioning, `v1.10` is considered less than `v1.9`. This adjusts the versioning to correctly sort the version numbers by major, minor and patch versions.

## Preview

Verify the `VersionBlock` components correctly show/hide based on the `firstVersion` & `lastVersion` props set:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/docs/version-testing

Verify this page hides from the sidebar for versions after `v1.8`:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Verify `Project configs` sidebar category is only visible for `v1.8` and up:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/dbt_project.yml

To quickly test the new `sortVersions` method, drop the following code into the versionBlock file and visit the `/version-testing` page above. There is a console.log within `sortVersions` which will show the sorted array of versions.
```
sortVersions([
  "1.11",
  "1.7",
  "2.1",
  "1.8",
  "1.6",
  "2.2.1",
  "1.10",
  "2.0.1",
  "1.9",
  "1.9.1",
  "1.10.1",
  "2.0",
]);
```

## TODO Before Merge
- [ ] Delete demo `version-testing` page
- [ ] Remove console.log from `sortedVersions`
- [ ] Revert versions in `dbt-versions`
- [ ] Remove demo `Project configs` versioned category in `dbt-versions`

mkottakota1 committed:
Update vertica-setup.md

dbt-core 1.8.3 release

mkottakota1 committed:
Update vertica-setup.md for DBT core 1.8.5

JKarlavige committed:
add method to sort versions and begin adding to versionBlock component

JKarlavige committed:
revert changes to cumulative-metrics page

JKarlavige committed:
sortVersions method working

diegoquintanav committed:
Update constraints.md

Adds a reference to `{{ target.schema }}`, useful when defining `foreign_key` constraints.

mirnawong1 committed:
update faq

mirnawong1 committed:
update faq

JKarlavige committed:
update versionBlock version handling

JKarlavige committed:
update page-version-check

JKarlavige committed:
create util to handle version check

JKarlavige committed:
update page-version-check to use util version check

JKarlavige committed:
update category-version-check to use version check util

JKarlavige committed:
remove custom variable component and variable file

mirnawong1 committed:
Update website/docs/reference/resource-properties/constraints.md

JKarlavige commented on contributing/single-sourcing-content.md:
Functionality not used, confirmed with Docs team we can remove the `Var` component: https://dbt-labs.slack.com/archives/C02NCQ9483C/p1726157412409069

JKarlavige commented on website/dbt-versions.js:
```diff
@@ -43,11 +55,11 @@ exports.versions = [
   {
-    "page": "/reference/resource-configs/target_database",
+    "page": "reference/resource-configs/target_database",
     "lastVersion": "1.8",
   },
   {
-    "page": "/reference/resource-configs/target_schema",
+    "page": "reference/resource-configs/target_schema",
```
Keep these changes, page paths should not have leading forward slash. Remove all other changes in this file.

JKarlavige commented on website/docs/docs/version-testing.md:
Delete this page before merge

JKarlavige commented on website/src/utils/sort-versions.js:
```diff
@@ -0,0 +1,46 @@
+    return 0
+  });
+
+  console.log('sortedVersions', sortedVersions)
```
Remove this console before merge

JKarlavige committed:
Merge branch 'current' into fix-version-orders

mirnawong1 committed:
Update website/docs/faqs/Troubleshooting/auth-expired-error.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/faqs/Troubleshooting/auth-expired-error.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/faqs/Troubleshooting/auth-expired-error.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>

mirnawong1 committed:
Update website/docs/reference/resource-properties/constraints.md

mirnawong1 committed:
replace w with r

nghi-ly committed:
Update Advanced CI overview and related pages (#6033)

## What are you changing in this pull request and why?

Reorganized the Advanced CI docs  

## Checklist
- [x] I have reviewed the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I
have versioned it according to the [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and/or [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
guidelines.
- [x] Needs PM review
- [x] Needs Legal review

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: nataliefiann <120089939+nataliefiann@users.noreply.github.com>

nghi-ly committed:
Fix card  (#6062)

## What are you changing in this pull request and why?

Fix card on landing page

## Checklist
- [x] I have reviewed the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I
have versioned it according to the [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and/or [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
guidelines.

john-rock commented:
Hey @JKarlavige nice work on this, qq on this page https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Confirmed it does get hidden from the sidebar after `v1.8` but is this the correct copy that is suppose to be showing if the page is accessed when the version is set > `v1.8`?
![image](https://github.com/user-attachments/assets/d8eb8caa-ee13-4335-81e1-d281ac5d5a53)

Specifically the "You should upgrade to 0 or later if you want to use this feature."

JKarlavige commented:
> Hey @JKarlavige nice work on this, qq on this page https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database
> 
> Confirmed it does get hidden from the sidebar after `v1.8` but is this the correct copy that is suppose to be showing if the page is accessed when the version is set > `v1.8`? ![image](https://private-user-images.githubusercontent.com/46692803/367398161-d8eb8caa-ee13-4335-81e1-d281ac5d5a53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNTE0OTIsIm5iZiI6MTcyNjI1MTE5MiwicGF0aCI6Ii80NjY5MjgwMy8zNjczOTgxNjEtZDhlYjhjYWEtZWUxMy00MzM1LTgxZTEtZDI4MWFjNWQ1YTUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEzVDE4MTMxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNDM0ZmRlNTA3NTM1YTk4NjcwN2Y5NzdhZTk1NzI2N2RkNzE2MjI5NjJhNjRlZTEzZGFjYjI1MzY4Mjk3OGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9e-j5JQomrCcuUXGcYglnY8oJGmpVCWGrlPmJF7Eeq4)
> 
> Specifically the "You should upgrade to 0 or later if you want to use this feature."

Nice catch, it should definitely not say that. Will work on fixing that!

mirnawong1 committed:
Update configure-auto-exposures.md

mirnawong1 committed:
Update auto-exposures.md

runleonarun committed:
Update airflow-and-dbt-cloud to new repo (#6064)

## What are you changing in this pull request and why?

* Updating to our own dbt maintained repo.

Note that Leah will update the videos next week.

## Checklist
- [ ] I have reviewed the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [ ] The topic I'm writing about is for specific dbt version(s) and I
have versioned it according to the [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and/or [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
guidelines.
- [ ] I have added checklist item(s) to this list for anything anything
that needs to happen before this PR is merged, such as "needs technical
review" or "change base branch."
<!--
PRE-RELEASE VERSION OF dbt (if so, uncomment):
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
<!-- 
ADDING OR REMOVING PAGES (if so, uncomment):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links
that point to deleted pages
-->

mirnawong1 committed:
Update website/docs/docs/collaborate/auto-exposures.md

mirnawong1 committed:
Update website/docs/docs/core/connect-data-platform/vertica-setup.md

mirnawong1 committed:
Update website/docs/docs/cloud-integrations/configure-auto-exposures.md

mirnawong1 committed:
Update configure-auto-exposures.md

adding per feedback: https://dbt-labs.slack.com/archives/C06GJRK27PF/p1726261331100539?thread_ts=1726163198.724299&cid=C06GJRK27PF

nataliefiann committed:
Update website/docs/docs/cloud-integrations/configure-auto-exposures.md

mirnawong1 committed:
Update website/docs/docs/collaborate/auto-exposures.md

JKarlavige committed:
update version banner messaging

JKarlavige committed:
Merge branch 'current' into fix-version-orders

JKarlavige committed:
remove check to always show banner if page not available

JKarlavige commented:
@john-rock Updated the version warning banner to exclude the "first available in version.." part of the message if a `firstVersion` prop isn't set for that versioned page. Previews below:

Verify version banner no longer shows incorrect upgrade recommendation line:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/resource-configs/target_database

Select `v1.7` or lower and verify version banner shows with correct message:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/reference/global-configs/indirect-selection

Verify all pages not listed in [versionedPages array](https://github.com/dbt-labs/docs.getdbt.com/blob/a3edaf54a1e02c989ff9917c96fb578d8e222731/website/dbt-versions.js#L44) do not show version warning banner:
https://docs-getdbt-com-git-fix-version-orders-dbt-labs.vercel.app/docs/introduction

john-rock commented on website/src/utils/sort-versions.js:
```diff
@@ -0,0 +1,46 @@
+    const bSegments = b?.split(".", 3);
+
+    // Store each version part in variable to help readability below
+    const aMajor = aSegments[0] || "0"
+    const bMajor = bSegments[0] || "0"
+    const aMinor = aSegments[1] || "0"
+    const bMinor = bSegments[1] || "0"
+    const aPatch = aSegments[2] || "0"
+    const bPatch = bSegments[2] || "0"
```
The lengths we go to with javascript 🤣 nice work on this util.

john-rock reviewed (approved):
This lgtm. I'm no longer seeing the incorrect warning message if `firstVersion` is not set.

JKarlavige committed:
revert testing edits

JKarlavige committed:
Merge branch 'current' into fix-version-orders
